crates/wasmparser/src/readers/core/types.rs

```diff
@@ -762,6 +881,75 @@ impl FuncType {
     }
 }
 
+impl Matches for FuncType {
+    fn matches<'a, F>(&self, other: &Self, type_at: &F) -> Result<bool>
+    where
+        F: Fn(u32) -> Result<&'a SubType>,
+    {
+        let r = self.params().len() == other.params().len()
+            && self.results().len() == other.results().len()
+            // Note: matching functions are contravariant in their parameter types.
```

Comment:
Could you expand this comment? For example this switches the order of the `matches` predicate which seems important to call out here, along with its implications.

Reply:
Adding a reference to the GC spec.